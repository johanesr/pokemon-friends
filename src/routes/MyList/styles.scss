.my-list-wrapper {
  .my-list-card-wrapper {
    display: grid;
    @media (min-width: 992px) {
      grid-template-columns: repeat(auto-fill, minmax(200px, 1fr));
    }
    @media (max-width: 992px) {
      grid-template-columns: repeat(auto-fill, minmax(100px, 1fr));
    }
    grid-gap: 20px;
    margin: 20px 50px;

    a{
      color: #000000;
      text-decoration: none;
    }

    .pokemon-list-card {
      border: 2px solid black;
      border-radius: 15px;
      background-color: lightgreen;
      text-align: center;
      cursor: pointer;
      transition: 0.3s linear;
      box-shadow: 5px 5px 5px #000000;
    }

    .pokemon-list-card:hover {
      animation-name: cardAnimation;
      animation-duration: 0.5s;
      animation-iteration-count: infinite;
      transform: scale(1.1,1.1);
    }

    .pokemon-name {
      font-weight: 700;
      padding: 20px 0px;
      color: #000000;
      text-transform: capitalize;
    }

    .pokemon-nickname {
      font-family: 'Hachi Maru Pop';
      margin: 15px 0px;

      span {
        padding-right: 10px;
      }
      button {
        border: 2px solid black;
        border-radius: 5px;
        color: white;
        background-color: red;
      }
    }

    img {
      height: 96px;
      width: 96px;
    }
  }
}