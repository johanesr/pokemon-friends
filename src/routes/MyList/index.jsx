import { useState, useEffect } from 'react';
import './styles.scss';

function MyList() {
  const [ownPokemon, setOwnPokemon] = useState([]);

  useEffect(() => {
    let keys = Object.keys(localStorage);
    let eachPokemon = {};
    let pokemons = [];

    for (let i=0; i< keys.length; i++) {
      try {
        eachPokemon = {
          name: keys[i],
          data: JSON.parse(localStorage.getItem(keys[i]))
        }
      } catch (e) {
        continue;
      }
      pokemons.push(eachPokemon);
      setOwnPokemon(pokemons);
    }
  }, [])

  function removePokemon(name, nickname) {
    const deletePokemon = JSON.parse(localStorage.getItem(name));
    let deleteNickname = deletePokemon.nickname;

    deleteNickname.splice(deleteNickname.indexOf(nickname),1);
    deletePokemon.nickname = deleteNickname;
    deletePokemon.count--;

    localStorage.removeItem(name);
    if(deletePokemon.nickname.length!==0) {
      localStorage.setItem(name, JSON.stringify(deletePokemon));
    }

    window.location.reload();
  }

  return (
    <div className="my-list-wrapper">
      <div className="my-list-card-wrapper">
        {ownPokemon.map((pokemon,i) => (
          <div className="pokemon-list-card" key={i}>
            <img src={pokemon.data.image} alt={pokemon.name}/>
            <div className="pokemon-name">{pokemon.name}</div>
            {pokemon.data.nickname.map((nickname, j) => (
              <div key={j} className="pokemon-nickname">
                <span>{nickname}</span>
                <button onClick={() => removePokemon(pokemon.name, nickname)}>Delete</button>
              </div>
            ))}
          </div>
        ))}
      </div>
    </div>
  )
}

export default MyList;