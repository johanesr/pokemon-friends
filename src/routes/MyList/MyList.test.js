import { mount } from 'enzyme';
import MyList from './index';

const wrapper = mount(<MyList />);

it("renders MyList Container correctly", () => {
  expect(wrapper).toMatchSnapshot(); // check if a container renders correctly
});