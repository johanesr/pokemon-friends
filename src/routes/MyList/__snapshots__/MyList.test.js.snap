// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MyList Container correctly 1`] = `
<MyList>
  <div
    className="my-list-wrapper"
  >
    <div
      className="my-list-card-wrapper"
    />
  </div>
</MyList>
`;
